htdocs/core/modules/societe/mod_codeclient_elephant.php

```diff
@@ -114,6 +117,33 @@ public function info($langs)
 		$texte .= '<td class="right">'.$form->textwithpicto('<input type="text" class="flat minwidth175" name="value2" value="'.getDolGlobalString('COMPANY_ELEPHANT_MASK_SUPPLIER').'"'.$disabled.'>', $tooltip, 1, 1).'</td>';
 		$texte .= '</tr>';
 
+		// Date of switch to that numberring model
+		$datedb = getDolGlobalString('COMPANY_ELEPHANT_DATE_START');
+		// After save, default dolibarr store data like displayed : 20/05/2024 and we need a timestamp -> override data
+		if (!empty($datedb)) {
+			if (!is_numeric($datedb) && GETPOSTISSET('value3')) {
+				if (GETPOST('value4') == 1) {
+					$dateinput = GETPOSTDATE('value3');
+					$res = dolibarr_set_const($this->db, 'COMPANY_ELEPHANT_DATE_START', $dateinput, 'chaine', 0, '', $conf->entity);
+				} else {
+					$res = dolibarr_set_const($this->db, 'COMPANY_ELEPHANT_DATE_START', '', 'chaine', 0, '', $conf->entity);
+				}
+			} else {
+				$dateinput = $datedb;
+			}
+		}
+		if (empty($dateinput)) {
+			$dateinput = dol_now();
+		}
+		$texte .= '<tr><td>'.$langs->trans("DateStartThatModel").' ('.$langs->trans("DateStartThatModelHelp").'):</td>';
+		$texte .= '<td class="" style="flex-direction: row;display: flex;">';
+		$texte .= '<input type="checkbox" onclick="let d=document.getElementById(\'elephantchoosedate\'); if(this.checked){d.style.cssText = \'display: block;\'}else{{d.style.cssText = \'display: none;\'}}" name="value4" value="1" style="float: left;"/>';
+		$texte .= '<div style="display: none;" id="elephantchoosedate" class="right">';
+		$texte .= $form->selectDate($dateinput, 'value3', 0, 0, 1, '', 1, 1, $disabled);
```

Comment:
`		$selectdisabled = ((!empty($mc->sharings['referent']) && $mc->sharings['referent'] != $conf->entity) ? 1 : 0);
		$texte .= $form->selectDate($dateinput, 'value3', 0, 0, 1, '', 1, 1, $selectdisabled);
`

Reply:
@frederic34 ... i really can't see that your code could do the same thing as mine ?

my objective is:
- add a checkbox and if user check it, then display the input field, uncheck that checkbox, hide the text input field
- on submit we could get value if checkbox is checked or no and then delete / update value

that is an elegant way to handle situation where people want to "remove" data: with classic input field if the field is empty on POST dolibarr do not "delete" / "set to empty" value (i think, that was the situation last time i check)...

Reply:
$disabled returns a string but selectDate wait for an int

Reply:
You probably want to do
$disabled ? 1 : 0